### PR TITLE
Refactor API code to base.py in RunInference

### PR DESCRIPTION
**Pull request by AnandInguva**
Refactor code from api to base.py. Delete api.py

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: "addresses #123), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment "fixes #<ISSUE NUMBER>" instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by AnandInguva** [comment]
R: @robertwb @ryanthompson591

File: sdks/python/apache_beam/ml/inference/sklearn_inference_test.py [hunk 1]
```diff
@@ -134,9 +133,9 @@ def test_predict_output(self):
         numpy.array([1, 2, 3]), numpy.array([4, 5, 6]), numpy.array([7, 8, 9])
     ]
     expected_predictions = [
-        api.PredictionResult(numpy.array([1, 2, 3]), 6),
-        api.PredictionResult(numpy.array([4, 5, 6]), 15),
-        api.PredictionResult(numpy.array([7, 8, 9]), 24)
+        base.PredictionResult(numpy.array([1, 2, 3]), 6),
```
Comment (ryanthompson591) [inline sdks/python/apache_beam/ml/inference/sklearn_inference_test.py:136]: WDYT of just importing PredictionResult directly like we do in the pytorch test.

**Review by ryanthompson591 — APPROVED** [review]
Looks good.
@ihji this will probably change what you do with your PR.

File: sdks/python/apache_beam/ml/inference/sklearn_inference_test.py (hunk 1 above)
Comment (AnandInguva) [reply sdks/python/apache_beam/ml/inference/sklearn_inference_test.py:136]: yep. we can do that.

Comment (AnandInguva) [reply sdks/python/apache_beam/ml/inference/sklearn_inference_test.py]: Done

File: sdks/python/apache_beam/ml/inference/base.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+# mypy: ignore-errors
```
Comment (yeandy) [inline sdks/python/apache_beam/ml/inference/base.py:17]: What was the error here? was it something like `Type variable is unbound`?

Comment (AnandInguva) [reply sdks/python/apache_beam/ml/inference/base.py:17]: Yes.

**Comment by ihji** [comment]
> Looks good. @ihji this will probably change what you do with your PR.

Got it. Will update my PR accordingly after this PR gets merged.

File: sdks/python/apache_beam/ml/inference/base.py (hunk 2 above)
Comment (tvalentyn) [reply sdks/python/apache_beam/ml/inference/base.py:17]: Do we understand why we hit that error?

**Comment by AnandInguva** [comment]
PTAL @tvalentyn

File: sdks/python/apache_beam/ml/inference/base.py (hunk 2 above)
Comment (tvalentyn) [reply sdks/python/apache_beam/ml/inference/base.py:18]: I think #21441  describes a different problem: supporting type inference for RunInference API users. I believe some of the approaches, that it mentions, have been  implemented (adding generics), and we have some type inference capabilities (that you plan to test in another test). I'd ask, what is the remaining work there?

The problem of having to disable mypy in this file seems to do with developer tooling. Do we know what causes mypy to complain? Does this problem need a separate Jira?

Comment (AnandInguva) [reply sdks/python/apache_beam/ml/inference/base.py:18]: `pache_beam/testing/load_tests/load_test_metrics_utils.py:51: error: unused 'type: ignore' comment
apache_beam/ml/inference/base.py:68: error: Type variable "apache_beam.ml.inference.base._INPUT_TYPE" is unbound  [valid-type]
apache_beam/ml/inference/base.py:68: note: (Hint: Use "Generic[_INPUT_TYPE]" or "Protocol[_INPUT_TYPE]" base class to bind "_INPUT_TYPE" inside a class)
apache_beam/ml/inference/base.py:68: note: (Hint: Use "_INPUT_TYPE" in function signature to bind "_INPUT_TYPE" inside a function)
apache_beam/ml/inference/base.py:69: error: Type variable "apache_beam.ml.inference.base._OUTPUT_TYPE" is unbound  [valid-type]
apache_beam/ml/inference/base.py:69: note: (Hint: Use "Generic[_OUTPUT_TYPE]" or "Protocol[_OUTPUT_TYPE]" base class to bind "_OUTPUT_TYPE" inside a class)
apache_beam/ml/inference/base.py:69: note: (Hint: Use "_OUTPUT_TYPE" in function signature to bind "_OUTPUT_TYPE" inside a function`

Comment (AnandInguva) [reply sdks/python/apache_beam/ml/inference/base.py:18]:
```
pache_beam/testing/load_tests/load_test_metrics_utils.py:51: error: unused 'type: ignore' comment
apache_beam/ml/inference/base.py:68: error: Type variable "apache_beam.ml.inference.base._INPUT_TYPE" is unbound  [valid-type]
apache_beam/ml/inference/base.py:68: note: (Hint: Use "Generic[_INPUT_TYPE]" or "Protocol[_INPUT_TYPE]" base class to bind "_INPUT_TYPE" inside a class)
apache_beam/ml/inference/base.py:68: note: (Hint: Use "_INPUT_TYPE" in function signature to bind "_INPUT_TYPE" inside a function)
apache_beam/ml/inference/base.py:69: error: Type variable "apache_beam.ml.inference.base._OUTPUT_TYPE" is unbound  [valid-type]
apache_beam/ml/inference/base.py:69: note: (Hint: Use "Generic[_OUTPUT_TYPE]" or "Protocol[_OUTPUT_TYPE]" base class to bind "_OUTPUT_TYPE" inside a class)
apache_beam/ml/inference/base.py:69: note: (Hint: Use "_OUTPUT_TYPE" in function signature to bind "_OUTPUT_TYPE" inside a function
```

File: sdks/python/apache_beam/ml/inference/base.py [hunk 3]
```diff
@@ -206,6 +216,19 @@ class RunInference(beam.PTransform[beam.PCollection[ExampleT],
...
+
+  A transform that takes a PCollection of examples (or features) to be used on
+  an ML model. It will then output inferences (or predictions) for those
+  examples in a PCollection of PredictionResults, containing the input examples
+  and output inferences.
+
+  If examples are paired with keys, it will output a tuple
+  (key, PredictionResult) for each (key, example) input.
+
+  Models for supported frameworks can be loaded via a URI. Supported services
```
Comment (tvalentyn) [inline sdks/python/apache_beam/ml/inference/base.py:228]: TODO in  a separate PR:
- proof-read these API bits by TW.
- Supported services can also be used - what is this about?
-   TODO(BEAM-14046): Add and link to help documentation : Let's point to the examples directory.

**Comment by tvalentyn** [comment]
merging based on results from previous commit run.

File: sdks/python/apache_beam/ml/inference/base.py (hunk 3 above)
Comment (AnandInguva) [reply sdks/python/apache_beam/ml/inference/base.py:229]: Yep. thanks

**Pull request merged**